library/core/src/num/uint_macros.rs

```diff
@@ -1115,6 +1115,36 @@ macro_rules! uint_impl {
             (a as Self, b)
         }
 
+        /// Calculates `self` + `rhs` + `carry`.
+        ///
+        /// Performs "ternary addition" which takes in an extra bit to add, and may return an
+        /// additional bit of overflow. This allows for chaining together multiple additions
+        /// to create "big integers" which represent larger values.
+        ///
+        /// # Examples
+        ///
+        /// Basic usage
+        ///
+        /// ```
+        /// #![feature(carrying)]
+        #[doc = concat!("assert_eq!(5", stringify!($SelfT), ".carrying_add(2, false), (7, false));")]
+        #[doc = concat!("assert_eq!(5", stringify!($SelfT), ".carrying_add(2, true), (8, false));")]
+        #[doc = concat!("assert_eq!(", stringify!($SelfT), "::MAX.carrying_add(1, false), (0, true));")]
+        #[doc = concat!("assert_eq!(", stringify!($SelfT), "::MAX.carrying_add(1, true), (1, true));")]
+        /// ```
+        #[unstable(feature = "carrying", issue = "none")]
+        #[rustc_const_unstable(feature = "const_carrying", issue = "none")]
+        #[must_use = "this returns the result of the operation, \
+                      without modifying the original"]
+        #[inline(always)]
+        pub const fn carrying_add(self, rhs: Self, carry: bool) -> (Self, bool) {
+            // note: longer-term this should be done via an intrinsic, but this has been shown
```

Comment:
You can consider following [the TODO convention](https://rustc-dev-guide.rust-lang.org/conventions.html#use-todo-comments-for-things-you-dont-want-to-forget).

Reply:
This says tidy won't allow the PR to be merged with it, though